traits/ctraits.c

```diff
@@ -4089,7 +4089,7 @@ _trait_rich_comparison ( trait_object * trait, PyObject * args ) {
 
     int compare_type;
 
-    if ( !PyArg_ParseTuple( args, "i", &compare_type ) )
+    if ( !PyArg_ParseTuple( args, "p", &compare_type ) )
```

Comment:
There's a pending PR from @rahulporuri that removes this function altogether; it may be worth leaving this change out of this PR to avoid the merge conflict. (OTOH, resolving the conflict shouldn't be hard, either.)

Reply:
On second thoughts, it should be no harder to fix #598 for the conflicts than to do a LBYL fix for this PR, so this shouldn't block merging.